DREAM3DReviewFilters/GenerateFeatureIDsbyBoundingBoxes.cpp

```diff
@@ -0,0 +1,514 @@
+    tempMod = i % (dims[0] * dims[1]);
+    yindex = tempMod / (dims[0]);
+    xindex = tempMod % (dims[0]);
+
+    currentx = origin[0] + spacing[0] * xindex;
+    currenty = origin[1] + spacing[1] * yindex;
+    currentz = origin[2] + spacing[2] * zindex;
+
+    inBounds = false;
+    k = 0;
+
+    while(!inBounds && k < totalNumFIDs)
+    {
+      xmin = m_BoxCenter[3 * k] - m_BoxDims[3 * k] / 2.0;
+      xmax = m_BoxCenter[3 * k] + m_BoxDims[3 * k] / 2.0;
+      ymin = m_BoxCenter[3 * k + 1] - m_BoxDims[3 * k + 1] / 2.0;
+      ymax = m_BoxCenter[3 * k + 1] + m_BoxDims[3 * k + 1] / 2.0;
+      zmin = m_BoxCenter[3 * k + 2] - m_BoxDims[3 * k + 2] / 2.0;
+      zmax = m_BoxCenter[3 * k + 2] + m_BoxDims[3 * k + 2] / 2.0;
+
+      inBounds = IsPointInBounds(xmax, xmin, ymax, ymin, zmax, zmin, currentx, currenty, currentz);
+
+      if(inBounds)
+      {
+        m_FeatureIds[i] = m_BoxFeatureIds[k];
+      }
+
+      k++;
+    }
+  }
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void GenerateFeatureIDsbyBoundingBoxes::checkBoundingBoxEdge()
+{
+  size_t totalNumFIDs = m_BoxFeatureIdsPtr.lock()->getNumberOfTuples();
+  size_t totalNumElementsDest = m_FeatureIdsPtr.lock()->getNumberOfTuples();
+
+  for(size_t i = 0; i < totalNumElementsDest; i++)
+  {
+    for(size_t k = 0; k < totalNumFIDs; k++)
+    {
+    }
+  }
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void GenerateFeatureIDsbyBoundingBoxes::checkBoundingBoxVertex()
+{
+  size_t totalNumFIDs = m_BoxFeatureIdsPtr.lock()->getNumberOfTuples();
+  size_t totalNumElementsDest = m_FeatureIdsPtr.lock()->getNumberOfTuples();
+  DataContainer::Pointer dc = getDataContainerArray()->getDataContainer(getFeatureIDsArrayPath().getDataContainerName());
+  VertexGeom::Pointer vertex = dc->getGeometryAs<VertexGeom>();
+
+  bool inBounds = false;
+  float xmin = 0;
+  float xmax = 0;
+  float ymin = 0;
+  float ymax = 0;
+  float zmin = 0;
+  float zmax = 0;
+
+  float currentx = 0;
+  float currenty = 0;
+  float currentz = 0;
+
+  float* vertices = vertex->getVertexPointer(0);
+  size_t k = 0;
+  for(size_t i = 0; i < totalNumElementsDest; i++)
+  {
+    inBounds = false;
+    k = 0;
+    currentx = vertices[3 * i];
+    currenty = vertices[3 * i + 1];
+    currentz = vertices[3 * i + 2];
+
+    while(!inBounds && k < totalNumFIDs)
+    {
+      xmin = m_BoxCenter[3 * k] - m_BoxDims[3 * k] / 2.0;
+      xmax = m_BoxCenter[3 * k] + m_BoxDims[3 * k] / 2.0;
+      ymin = m_BoxCenter[3 * k + 1] - m_BoxDims[3 * k + 1] / 2.0;
+      ymax = m_BoxCenter[3 * k + 1] + m_BoxDims[3 * k + 1] / 2.0;
+      zmin = m_BoxCenter[3 * k + 2] - m_BoxDims[3 * k + 2] / 2.0;
+      zmax = m_BoxCenter[3 * k + 2] + m_BoxDims[3 * k + 2] / 2.0;
```

Comment:
```suggestion
  float* vertices = vertex->getVertexPointer(0);
  for(size_t i = 0; i < totalNumElementsDest; i++)
  {
    bool inBounds = false;
    size_t k = 0;
    float currentx = vertices[3 * i];
    float currenty = vertices[3 * i + 1];
    float currentz = vertices[3 * i + 2];

    while(!inBounds && k < totalNumFIDs)
    {
      float xmin = m_BoxCenter[3 * k] - m_BoxDims[3 * k] / 2.0;
      float xmax = m_BoxCenter[3 * k] + m_BoxDims[3 * k] / 2.0;
      float ymin = m_BoxCenter[3 * k + 1] - m_BoxDims[3 * k + 1] / 2.0;
      float ymax = m_BoxCenter[3 * k + 1] + m_BoxDims[3 * k + 1] / 2.0;
      float zmin = m_BoxCenter[3 * k + 2] - m_BoxDims[3 * k + 2] / 2.0;
      float zmax = m_BoxCenter[3 * k + 2] + m_BoxDims[3 * k + 2] / 2.0;
```
Move variable declarations inside the loop.